### ISSUE TITLE
Prep for Diva6 release

### ISSUE THREAD

**Issue by EricHanLiu** [DDMAL/diva.js#406]
## To-do list
- [x] Unit tests
  - [x] Test they work
  - [x] Update them
    - [x] issue #409 
    - [x] issue #412 
    - [ ] issue #422 (increase code coverage, write tests for plugins)
  - [x] issue #444 UNIT TESTS FAILING (related to issue #441)
- [x] Find and fix any outstanding bugs
  - [x] issue #350 
  - [x] issue #392 
  - [x] issue #376 
  - [x] issue #365 
  - [x] issue #344 
  - [x] issue #324 
  - [x] issue #400 
  - [x] issue #416 
  - [x] issue #425 (for metadata plugin)
  - [x] issue #436
  - [x] issue #442 
  - [x] Page Label / Visible Pages stuff 
    - [x] issue #434 
    - [x] issue #404 
    - [x] issue #443 
  - [x] issue #446 
  - [x] issue #447
  - [x] issue #380 - rendering issue, valid?
  - [x] issue #441 - viewport size change
- [x] Update documentation 
  - [x] issue #413 
  - [x] issue #414 (README)
  - [x] issue #431, update wiki and website
- [x] Update dependencies 
  - [x] Vulnerable
  - [x] Outdated: issue #410
- [x] New features
  - [x] issue #411 
  - [x] issue #335 
  - [x] issue #418 
  - [x] issue #421 
  - [x] issue #375 Manipulation Plugin
    - [x] issue #438 Add reset
    - [x] issue #439 Secondary pages only once
    - [x] issue #440 Make zoomable
    - [x] issue #445 Sharpness Filter
    - [x] issue #448 Threshold filter
    - [x] issue #450 rotate
    - [x] issue #451 dbl click zoom
    - [x] issue #452 close button
    - [x] issue #453 mirror controls
    - [x] issue #449 new filters
    - [x] issue #461 mobile
  - [x] issue #432 highlight plugin - NOT FOR INITIAL RELEASE
  - [x] issue #415 secondary images tool - NOT FOR INITIAL RELEASE
- [ ] Final preparations for release (after other issues are done)
  - [ ] issue #374, add to CDNJS (should only be done after the release is ready)
  - [x] add latest Diva to npm
  - [x] create release package
  - [x] update release packages on site and wiki

PR #437 contains a lot of changes, accepted!

**Comment by EricHanLiu** [comment]
Fixed JSHint syntax issues and added proper Babel preset for Webpack compiling

**Comment by EricHanLiu** [comment]
None of the unit tests work on `master` or `develop-diva6`

**Comment by EricHanLiu** [comment]
Dependencies updated
```json
gulp -> 4.0.0
webpack -> 4.14.0
webpack-dev-server -> 3.1.4
mocha -> 5.2.0
babel-loader -> 7.1.4
```

**Comment by ahankinson** [comment]
As this release is imminent:

1. Merge your commits on this branch back to `develop-diva6`
2. Merge those commits from `develop-diva6` to `develop`
3. Base your changes going forward on the `develop` branch.

There is no need to keep so many branches around.

**Comment by EricHanLiu** [comment]
Everything merged into develop

**Comment by ahankinson** [comment]
OK. You should probably break out the other checkboxed to-dos into specific issues.

**Comment by EricHanLiu** [comment]
reminder to fix event listener issue on fullscreen mode

**Comment by EricHanLiu** [comment]
Issue #434 and #404 are awaiting PR #435 

Issue #375 is being done on the [filter queue](https://github.com/DDMAL/diva.js/tree/filter_queue) branch

**Comment by EricHanLiu** [comment]
Release should be ready

**Comment by EricHanLiu** [comment]
This isn't being automatically added to CDNJS based on the git release tag (which it says it should be doing)

**Comment by EricHanLiu** [comment]
Closing this in favour of the existing open issues for CDNJS and cantus ultimus testing. Other issues will be relative to a new minor release
